### DISCUSSION TITLE
Enchainement des épisodes sur ianime

### DISCUSSION THREAD

**Discussion by Ayuzerak** [Kodi-vStream/venom-xbmc-addons#4247] [Support & aide]
L'enchainement des épisodes avec upnext ne se fait pas sur ianime, quelles changements faire dans ianime.py ?

**Comment by sizanic** [comment]
Bonjour,

l'enchainement est géré par la classe upnext, ici :
https://github.com/Kodi-vStream/venom-xbmc-addons/blob/667db56b71a1cc8637a2e3bf2919e6ec7e19d708/plugin.video.vstream/resources/lib/upnext.py#L176

La fonction parcourt les dossiers qui contient les épisodes à la recherche de celui qui a le numéro suivant.
Le numéro de l'épisode est déterminé à partir du titre : 
https://github.com/Kodi-vStream/venom-xbmc-addons/blob/667db56b71a1cc8637a2e3bf2919e6ec7e19d708/plugin.video.vstream/resources/lib/gui/guiElement.py#L243

Le code est générique,
il faut donc que dans ianime, le titre soit bien formaté avec la saison et l'épisode.
https://github.com/Kodi-vStream/venom-xbmc-addons/blob/667db56b71a1cc8637a2e3bf2919e6ec7e19d708/plugin.video.vstream/resources/sites/ianime.py#L639

**Comment by Ayuzerak** [comment]
J'ai réussi à le faire sur les autres sites mais là ça coince. Je verrai plus tard.

**Comment by sizanic** [comment]
Quels autres sites ?
quelles modifications ?

**Reply by Ayuzerak** [reply]
Sur dulourd, 1.seriestreaming, et sur wiflix (au delà des saisons 1 pour ce site) cela ne fonctionnait pas. J'ai fait comme d'habitude j'ai indiqué dans le titre la saison et l'episode selon le format attendu mais rien ne se passe pour ianime.
